lib/HttpLightningBackend.js

```diff
@@ -37,12 +37,14 @@ class HttpLightningBackend extends LightningBackend {
 			}
 			const requestUrl = `${baseUrl}${uri}`;
 			const parsedUrl = url.parse(requestUrl);
+			const finalHeaders = Object.assign(Object.assign({}, headers), extraHeaders);
 			let options = {
 				method: method.toUpperCase(),
 				hostname: parsedUrl.hostname,
 				port: parsedUrl.port,
 				path: parsedUrl.path,
-				headers: Object.assign({}, headers),
+				 // extraHeaders are specific to a single request, and therefore should not be set on the class-scope "options" object.
+				headers: finalHeaders,
```

Comment:
^ that was my thinking at least, but i could be misunderstanding how these classes work